cwms-data-api/src/main/java/cwms/cda/data/dao/timeseriesprofile/TimeSeriesProfileInstanceDao.java

```diff
@@ -0,0 +1,706 @@
+                } else {
+                    List<TimeSeriesData> dataList = new ArrayList<>();
+                    dataList.add(new TimeSeriesData(dataValue.getValue().getVALUE(),
+                            dataValue.getValue().getQUALITY_CODE().intValue()));
+                    timeSeriesProfileInstanceList.put(dateTime.getTime(), dataList);
+                }
+            }
+        }
+
+        // add null values to the TimeSeriesProfileInstance value list if
+        // the data is missing for the associated parameter
+        for (Map.Entry<Long, List<TimeSeriesData>> entry : timeSeriesProfileInstanceList.entrySet()) {
+            if (entry.getValue().size() < paramList.size()) {
+                for (int i = 0; i < paramList.size(); i++) {
+                    Timestamp dateTime = Timestamp.from(Instant.ofEpochMilli(entry.getKey()));
+                    try {
+                        entry.getValue().get(i);
+                    } catch (IndexOutOfBoundsException e) {
+                        timeSeriesProfileInstanceList.get(dateTime.getTime()).add(i, null);
+                        continue;
+                    }
+                    if (timeValuePairMap.get(dateTime) == null
+                            || !timeValuePairMap.get(dateTime).containsKey(paramList.get(i).getParameter())) {
+                        timeSeriesProfileInstanceList.get(dateTime.getTime()).add(i, null);
+                    }
+                }
+            }
+        }
+
+        STR_TAB_T units = new STR_TAB_T(unit);
+
+        TS_PROF_DATA_T timeSeriesProfileData = new TS_PROF_DATA_T(locationCode, keyParameterCode,
+                timeZone, units, records);
+
+        // Get reference timeseries ID
+        AV_TS_PROFILE profileView = AV_TS_PROFILE.AV_TS_PROFILE;
+        CwmsId tsCwmsId = null;
+        try {
+            Long tsId = Objects.requireNonNull(dsl.selectDistinct(profileView.REFERENCE_TS_CODE)
+                    .from(profileView)
+                    .join(VIEW)
+                    .on(profileView.LOCATION_CODE.eq(VIEW.LOCATION_CODE)
+                            .and(profileView.KEY_PARAMETER_CODE.eq(VIEW.KEY_PARAMETER_CODE))
+                            .and(profileView.OFFICE_ID.eq(VIEW.OFFICE_ID)))
+                    .where(profileView.KEY_PARAMETER_ID.eq(keyParameter)
+                            .and(profileView.LOCATION_ID.eq(location.getName())
+                                    .and(profileView.OFFICE_ID.eq(location.getOfficeId()))))
+                    .fetchOne()).value1();
+            if (tsId != null) {
+                tsCwmsId = CwmsId.buildCwmsId(CWMS_TS_PACKAGE.call_GET_DB_OFFICE_ID(dsl.configuration(), tsId),
+                        CWMS_TS_PACKAGE.call_GET_TS_ID(dsl.configuration(), tsId));
+            }
+        } catch (NullPointerException e) {
+            LOGGER.log(Level.CONFIG, "No reference time series found for time series profile instance");
```

Comment:
don't catch null pointer exceptions

Reply:
Removed null pointer catch and replaced with a null check